go/vt/vtgate/vcursor_impl.go

```diff
@@ -211,6 +211,11 @@ func (vc *vcursorImpl) TimeZone() *time.Location {
 	return vc.safeSession.TimeZone()
 }
 
+func (vc *vcursorImpl) AllowZeroDate() bool {
+	// TODO: Implement detecting zero date
```

Comment:
@harshit-gangal Do you have a recommendation for how we can implement this here?

Reply:
We would need sql_mode representation which can provide these answers